client/web-sveltekit/src/testing/integration.ts

```diff
@@ -177,19 +168,29 @@ class Sourcegraph {
             }
         })
 
-        return async () => {
-            // Wait for the search results page to be "ready"
-            await this.page.getByRole('heading', { name: 'Filter results' }).waitFor()
-            return this.page.evaluate(
-                ([events]) => {
-                    for (const event of events) {
-                        for (const source of window.$$sources) {
-                            source.dispatchEvent(new MessageEvent(event.event, { data: JSON.stringify(event.data) }))
+        return {
+            publish: async (events: SearchEvent | SearchEvent[]): Promise<void> => {
```

Comment:
I'd make this a variadic function

```suggestion
            publish: async (...events: SearchEvent[]): Promise<void> => {
```

then it can be used in the following ways:

```
publish(event) // single event
publish(event1, event2) // multiple events
publish(...events) // array of events
```

Reply:
Oh, much more elegant!